master/internal/stream/model_versions.go

```diff
@@ -0,0 +1,256 @@
+package stream
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/stream"
+	"github.com/determined-ai/determined/proto/pkg/modelv1"
+)
+
+const (
+	// ModelVersionsDeleteKey specifies the key for delete model_versions.
+	ModelVersionsDeleteKey = "modelversions_deleted"
+	// ModelVersionsUpsertKey specifies the key for upsert model_versions.
+	ModelVersionsUpsertKey = "modelversion"
+	// model_versionChannel specifies the channel to listen to model_version events.
+	modelVersionChannel = "stream_model_version_chan"
+)
+
+// ModelVersionMsg is a stream.Msg.
+//
+// determined:stream-gen source=server delete_msg=ModelVersionsDeleted
+type ModelVersionMsg struct {
+	bun.BaseModel `bun:"table:model_versions"`
+
+	// immutable attributes
+	ID int `bun:"id,pk" json:"id"`
+
+	// mutable attributes
+	Name            string    `bun:"name" json:"name"`
+	Version         int       `bun:"version" json:"version"`
+	CheckpointUUID  string    `bun:"checkpoint_uuid" json:"checkpoint_uuid"`
+	CreationTime    time.Time `bun:"creation_time" json:"creation_time"`
+	LastUpdatedTime time.Time `bun:"last_updated_time" json:"last_updated_time"`
+	Metadata        JSONB     `bun:"metadata,type:jsonb" json:"metadata"`
+	ModelID         int       `bun:"model_id" json:"model_id"`
+	UserID          int       `bun:"user_id" json:"user_id"`
+	Comment         string    `bun:"comment" json:"comment"`
+	Labels          []string  `bun:"labels,array" json:"labels"`
+	Notes           string    `bun:"notes" json:"notes"`
+	// metadata
+	Seq int64 `bun:"seq" json:"seq"`
```

Comment:
For RBAC purposes, we'd typically want to include the relevant workspace_ids associate with each entity we're streaming, since we typically define permissions scopes based on the workspace. Do we know if RBAC-scoping around `ModelVersions` are workspace-focused?

If so we'd likely want to include the workspace_id as field in ModelVersionMsg for permission filtering